### PR TITLE
bugfix/#5600/#5602/Fixed the event filters.

### PR DESCRIPTION
**Pull request by Tadei-Mykola**

### CODE REVIEW

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 1]
```diff
@@ -103,34 +115,76 @@ export class EventsListComponent implements OnInit, OnDestroy {
       });
   }
 
-  updateSelectedFilters(value: any, event): void {
+  public updateSelectedFilters(value: any, event, optionsList?: any, dropdownName?: string, filterList?: any): void {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: don't use any type

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 2]
```diff
@@ -103,34 +115,76 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
+    if (value.nameEn === 'Upcoming') {
+      value.nameEn = 'Future';
+    } else if (value.nameEn === 'Passed') {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: use short if statements here

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 3]
```diff
@@ -103,34 +115,76 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
+      value.nameEn = 'Past';
+    }
     const existingFilterIndex = this.selectedFilters.indexOf(value);
     if (event.isUserInput && !event.source.selected && existingFilterIndex !== -1) {
       this.selectedFilters.splice(existingFilterIndex, 1);
+      this.deleteFromEventFilterCriteria(value, dropdownName);
+      this.checkAllSelectedFilters(value, optionsList, dropdownName, filterList);
+    } else if (event.isUserInput && !event.source.selected) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:127]: the same

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 4]
```diff
@@ -103,34 +115,76 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
-      { control: this.timeFilterControl },
-      { control: this.locationFilterControl },
-      { control: this.statusFilterControl },
-      { control: this.typeFilterControl }
-    ];
-
-    formControls.forEach((formControl) => {
-      formControl.control.valueChanges.subscribe((value) => {
-        this.updateSelectedFilters(value, Event);
+  private checkAllSelectedFilters(value: any, optionsList: any, dropdownName: string, filterList: any) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: fix any types

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 5]
```diff
@@ -179,11 +230,32 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
-      if (item) {
+    let deleted = true;
+    const keyMapping = {
+      eventTime: 0,
+      statuses: 1,
+      cities: 2,
+      tags: 3
+    };
+    [this.timeList, this.statusesList, this.locationList, this.typesList].forEach((list, arrayIndex) => {
+      for (const dropdownName in this.allSelectedFilter) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: use Object.keys(this.allSelectedFilter).forEach

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 6]
```diff
@@ -195,6 +267,16 @@ export class EventsListComponent implements OnInit, OnDestroy {
       });
     });
     this.selectedFilters.splice(0, this.selectedFilters.length);
+    for (const criteria in this.eventFilterCriteria) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: the same

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 7]
```diff
@@ -195,6 +267,16 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
+      if (this.eventFilterCriteria.hasOwnProperty(criteria)) {
+        this.eventFilterCriteria[criteria] = [];
+      }
+    }
+    for (const dropdownName in this.allSelectedFlags) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: the same

**Review by KKatiaa — APPROVED** [review]

**Review by sobkiv — APPROVED** [review]

**Pull request closed without merging**